From 9b6cd8c3c98283404d1ba6f64b7a05e15ebd7b1d Mon Sep 17 00:00:00 2001
From: Krystian Zlomek <k.zlomek@eyeo.com>
Date: Fri, 30 May 2025 14:12:03 +0000
Subject: [PATCH] DPD-1744: Disabled notifications in incognito due to crash

Assert due to invalid sequence:
```
```

To be investigated and enabled back under different ticket.

Signed-off-by: Krystian Zlomek <k.zlomek@eyeo.com>


(cherry picked from commit 93f4fcd26197920e7c71203ada029056fc68ab6f)

Co-authored-by: Krystian Zlomek <k.zlomek@eyeo.com>
---
 components/adblock/content/browser/element_hide_api_handler.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/components/adblock/content/browser/element_hide_api_handler.cc b/components/adblock/content/browser/element_hide_api_handler.cc
index 419f45ea014e9..b7ef483fccee4 100644
--- a/components/adblock/content/browser/element_hide_api_handler.cc
+++ b/components/adblock/content/browser/element_hide_api_handler.cc
@@ -60,6 +60,9 @@ void ElementHideApiHandler::LogSelectors(
     // The frame has been destroyed, so we can't notify observers.
     return;
   }
+  if (rfh->GetBrowserContext()->IsOffTheRecord()) {
+    return;
+  }
   auto* rcr = static_cast<ResourceClassificationRunnerImpl*>(
       ResourceClassificationRunnerFactory::GetForBrowserContext(
           rfh->GetBrowserContext()));
-- 
GitLab

